### PR TITLE
8.0 IMP purch_order: allow to using xml with no pdf and hook to customize xml

### PR DESCRIPTION
**Pull request by bealdav**
UBL is not always used with pdf but also with EDI plateform with xml only

Here some function split to serve multi usage

Please @astirpe @lmignon @SimoRubi @alexis-via could you have a look here.

Thanks a lot for your contribution

### CODE REVIEW

**Review by astirpe — APPROVED** [review]
LGTM

File: purchase_order_ubl/models/purchase.py [hunk 1]
```diff
@@ -206,16 +220,23 @@ def get_ubl_lang(self):
...
-        elif self.state in self.get_order_states():
-            doc_type = 'order'
+        doc_type = self.guess_ubl_doc_type()
         if doc_type:
             version = self.get_ubl_version()
             ubl_filename = self.get_ubl_filename(doc_type, version=version)
             xml_string = self.generate_ubl_xml_string(
                 doc_type, version=version)
+            self._ubl_check_xml_schema(
+                xml_string, self.document_name(doc_type), version=version)
```
Comment (astirpe) [inline purchase_order_ubl/models/purchase.py]: Here you invoke `document_name` but method name is `get_document_name`

File: purchase_order_ubl/models/purchase.py [hunk 2]
```diff
@@ -206,16 +220,23 @@ def get_ubl_lang(self):
...
             pdf_content = self.embed_xml_in_pdf(
                 xml_string, ubl_filename, pdf_content)
         return pdf_content
+
+    def guess_ubl_doc_type(self):
```
Comment (astirpe) [inline purchase_order_ubl/models/purchase.py]: Could you rename this method into `get_ubl_purchase_order_doc_type`? To make it consistent with https://github.com/OCA/edi/commit/4332084640b1933e6cf3742d4d697f2d2e8839a2#diff-54f706660be5cb2e069c3ec2c09f6dd9R214

**Review by astirpe — CHANGES_REQUESTED** [review]

File: purchase_order_ubl/models/purchase.py [hunk 3]
```diff
@@ -171,20 +188,17 @@ def generate_ubl_xml_string(self, doc_type, version='2.1'):
...
-            xml_declaration=True)
-        self._ubl_check_xml_schema(xml_string, document, version=version)
-        logger.debug(
-            '%s UBL XML file generated for purchase order ID %d (state %s)',
-            doc_type, self.id, self.state)
-        logger.debug(xml_string)
-        return xml_string
+        return xml_root
+
+    def get_document_name(self, doc_type):
```
Comment (astirpe) [inline purchase_order_ubl/models/purchase.py:196]:
```suggestion
    def get_document_name(self, doc_type):
        document = False
```

**Review by astirpe — APPROVED** [review]

**Comment by bealdav** [comment]
Thanks a lot @astirpe

**Review by florian-dacosta — APPROVED** [review]
LGTM

**Review by bguillot — APPROVED** [review]
LGTM

**Comment by bealdav** [comment]
Hi @alexis-via you're the man of the situation here. Thanks

**Comment by bealdav** [comment]
Thanks a lot Alexis

**Pull request merged**